File: app/src/mock/java/com/kirchhoff/example/githubclient/repository/FakeKeyValueStorage.java
package com.kirchhoff.example.githubclient.repository;

import android.support.annotation.NonNull;

import com.kirchhoff.example.githubclient.repository.keyvalue.KeyValueStorage;

/**
 * @author Kirchhoff-
 */

public class FakeKeyValueStorage implements KeyValueStorage {

    private String userName;

    public FakeKeyValueStorage() {
    }


    @Override
    public void saveToken(@NonNull String token) {

    }

    @NonNull
    @Override
    public String getToken() {
        return null;
    }

    @Override
    public void saveUserName(@NonNull String userName) {
        this.userName = userName;
    }

    @NonNull
    @Override
    public String getUserName() {
        return userName;
    }

    @Override
    public void clear() {

    }
}

File: app/src/mock/java/com/kirchhoff/example/githubclient/repository/FakeKeyValueStorage.kt
package com.kirchhoff.example.githubclient.repository

import com.kirchhoff.example.githubclient.repository.keyvalue.KeyValueStorage

/**
 * @author Kirchhoff-
 */
object FakeKeyValueStorage : KeyValueStorage {

    private var fakeUserName = ""

    override fun saveToken(token: String) {
        //Empty
    }

    override val token: String
        get() = ""

    override fun saveUserName(userName: String) {
        fakeUserName = userName
    }

    override val userName: String
        get() = fakeUserName

    override fun clear() {
        //Empty
    }

}